scripts/push_sdk/tests/test_push_php_sdk.py

```diff
@@ -0,0 +1,183 @@
+
+    self.os_client_builder = (OsClientBuilder().that_responds_on_get_generated_sources_base_path(self.generated_sources)
+                              .that_responds_on_get_sdk_repo_base_path(self.sdk_repository))
+
+  def test_execute_should_fail_when_sdks_folder_not_found(self):
+    # Arrange
+    fs_client = self.fs_client_builder.that_responds_on_exists(self.generated_sources, False).client
+    pipeline = PushPhpSdkAction(self.git_client_builder.client, fs_client, self.os_client_builder.client)
+
+    # Act & Assert
+    with pytest.raises(FileNotFoundError):
+      pipeline.execute()
+  
+  
+  def test_execute_should_fail_when_invalid_folder_name_for_generate_source(self):
+    # Arrange
+    invalid_folder_name = 'invalid_folder_name'
+    
+    fs_client = (self.fs_client_builder.that_responds_on_list_dir(self.generated_sources, [invalid_folder_name])
+      .client)
+    pipeline = PushPhpSdkAction(self.git_client_builder.client, fs_client, self.os_client_builder.client)
+    
+    # Act & Assert
+    with pytest.raises(InvalidCriteoServiceException):
+      pipeline.execute()
+  
+  def test_execute_should_fail_when_invalid_criteo_service(self):
+    # Arrange
+    invalid_criteo_service = 'invalid_criteo_service'
+    
+    fs_client = self.fs_client_builder.that_responds_on_list_dir(self.generated_sources, [f'{invalid_criteo_service}_preview']).client
+    pipeline = PushPhpSdkAction(self.git_client_builder.client, fs_client, self.os_client_builder.client)
+    
+    # Act & Assert
+    with pytest.raises(InvalidCriteoServiceException):
+      pipeline.execute()
+      
+  def test_execute_should_fail_when_invalid_api_version(self):
+    # Arrange
+    invalid_api_version = 'invalid-api-version'
+    
+    fs_client = self.fs_client_builder.that_responds_on_list_dir(self.generated_sources, [f'{CriteoService.marketingsolutions}_{invalid_api_version}']).client
+    pipeline = PushPhpSdkAction(self.git_client_builder.client, fs_client, self.os_client_builder.client)
+    
+    # Act & Assert
+    with pytest.raises(InvalidApiVersionException):
+      pipeline.execute()
+  
+  def test_execute_should_fail_when_remove_previous_sources_fails(self):
+    # Arrange
+    expected_exception = FileNotFoundError()
+    
+    fs_client = self.fs_client_builder.that_fails_on_remove(expected_exception).client
+    pipeline = PushPhpSdkAction(self.git_client_builder.client, fs_client, self.os_client_builder.client)
+    
+    # Act & Assert
+    with pytest.raises(Exception):
+      pipeline.execute()
+  
+  def test_execute_should_fail_when_copy_new_sources_fails(self):
+    # Arrange
+    expected_exception = FileNotFoundError()
+    
+    fs_client = self.fs_client_builder.that_fails_on_copy(expected_exception).client
+    pipeline = PushPhpSdkAction(self.git_client_builder.client, fs_client, self.os_client_builder.client)
+    
+    # Act & Assert
+    with pytest.raises(Exception):
```

Comment:
Just curious - are you able to check that it's a FileNotFoundError as well?

Reply:
Yes it should check for a FileNotFoundError, I'll change that thanks